### ISSUE TITLE
Terminating whole response on error in one of the backends

### ISSUE THREAD

**Issue by tosikszymon** [devopsfaith/krakend-ce#379]
Hi, I am wondering how potentially could be implemented scenario where based on error in one of the backend I terminate the whole pipe with some unified error code indicating what went wrong. Below is an example of how the the following flow is constructed:

```
                        API - 2
                       /            \
API-1 ---->                          Response 
                       \            /
                        API - 3 
```

First Step : Call API-1
Second Step : Call API-2, API-3 in parallel
Third Step : Return response

What I would like to achieve is in case where API-2 is failing with 4xx/5xx to terminate the whole request and return unified error, same for API-3. Even if it was sequential scenario like below:

`API-1 -> API-2 -> API-3 -> Response `

I ended up with having part of expected properties mapped from API-2 correctly but API-3 which failed was initiated with error code etc. 

```
{
	"error_backend_b": {
        "http_status_code": 404,
        "http_body": "404 page not found\\n"
	},
	"foo": 42
}
```
But for me personally it should either succeed or fail, not partially return something which as a whole does not bring any value.

What's the best way to go around this?

**Comment by kpacha** [comment]
hi @tosikszymon 

TLDR; no, there is no out-of-the-box way to enforce an all-or-nothing strategy for composing responses, but you can inject a custom plugin and replace the incomplete responses with whatever you want.

In our opinion, gateways and other parts of any µservice environment should support graceful degradation instead of amplificate every isolated error to the entire stack. That's why krakend returns incomplete responses to the client when an error occurred, letting the client decide what to do with the data the gateway was able to gather (or, as you put it, determine the value of the returned data). Checkout and splash screens have very different requirements regarding completeness...

I think an example will help, so let's say you have an endpoint consuming 3 different backend services, each with a 10% error rate. If we allow incomplete responses, the final error rate for the endpoint is `0.1*0.1*0.1=0.1%` (all 3 backends must fail at the same time). If we enforce an all-or-nothing strategy, the final error rate is `1-((1-0.1)*(1-0.1)*(1-0.1))=27.1%` (all the scenarios other than the one where the 3 services do not fail)

cheers
